src/mixer/main.go

```diff
@@ -11,143 +11,193 @@ import (
+	if err != nil {
+		os.Exit(-1)
 	}
 
-	if buildall.Parsed() {
-		b := builder.NewFromConfig(*buildallconf)
-		rpms, err := ioutil.ReadDir(b.Rpmdir)
-		if err == nil {
-			b.AddRPMList(rpms)
-		}
-		BuildChroots(b, *certflag)
-		BuildUpdate(b, *prefixflag, *minvflag, *formatflag, *signflag, !(*publishflag), *keepchrootsflag)
+	if v.Increment {
 		b.UpdateMixVer()
 	}
+}
 
-	if bundlescmd.Parsed() {
-		b := builder.NewFromConfig(*bundleconf)
-		fmt.Println("Getting clr-bundles for version " + b.Get("Clearver"))
-		b.UpdateRepo(b.Get("Clearver"), false)
-	}
+func cmdBuildImage(args []string) {
+	imagecmd := flag.NewFlagSet("build-image", flag.ExitOnError)
+	imageformat := imagecmd.String("format", "", "Supply the format used for the Mix")
 
-	if initcmd.Parsed() {
-		b := builder.New()
-		b.LoadBuilderConf(*initconf)
-		b.ReadBuilderConf()
-		b.InitMix(strconv.Itoa(*clearflag), strconv.Itoa(*mixflag), *allflag)
-	}
+	imagecmd.Parse(args)
 
-	if chrootcmd.Parsed() {
-		b := builder.NewFromConfig(*chrootconf)
-		BuildChroots(b, *certflag)
-	}
+	b := builder.NewFromConfig("")
+	b.BuildImage(*imageformat)
+}
 
-	if updatecmd.Parsed() {
-		b := builder.NewFromConfig(*updateconf)
-		BuildUpdate(b, *prefixflag, *minvflag, *formatflag, *signflag, !(*publishflag), *keepchrootsflag)
+func cmdAddRPMs(args []string) {
+	flags := flag.NewFlagSet("add-rpms", flag.ExitOnError)
+	conf := flags.String("config", "", "Supply a specific builder.conf to use for mixing")
+	flags.Parse(args)
 
-		if *incrementflag == true {
-			b.UpdateMixVer()
-		}
+	b := builder.NewFromConfig(*conf)
+	rpms, err := ioutil.ReadDir(b.Rpmdir)
+	if err != nil {
+		fmt.Printf("ERROR: cannot read %s\n", b.Rpmdir)
 	}
+	b.AddRPMList(rpms)
+}
 
-	if imagecmd.Parsed() {
-		b := builder.NewFromConfig("")
-		b.BuildImage(*imageformat)
-	}
+func cmdGetBundles(args []string) {
+	bundlescmd := flag.NewFlagSet("get-bundles", flag.ExitOnError)
+	bundleconf := bundlescmd.String("config", "", "Supply a specific builder.conf to use for mixing")
+	bundlescmd.Parse(args)
+	b := builder.NewFromConfig(*bundleconf)
+	fmt.Println("Getting clr-bundles for version " + b.Get("Clearver"))
+	b.UpdateRepo(b.Get("Clearver"), false)
+}
+
+func cmdInitMix(args []string) {
+	initcmd := flag.NewFlagSet("init-mix", flag.ExitOnError)
+	allflag := initcmd.Bool("all", false, "Create a mix with all Clear bundles included")
+	clearflag := initcmd.Int("clearver", 0, "Supply the Clear version to compose the mix from")
+	mixflag := initcmd.Int("mixver", 0, "Supply the Mix version to build")
+	initconf := initcmd.String("config", "", "Supply a specific builder.conf to use for mixing")
+	initcmd.Parse(args)
+	b := builder.New()
+	b.LoadBuilderConf(*initconf)
+	b.ReadBuilderConf()
+	b.InitMix(strconv.Itoa(*clearflag), strconv.Itoa(*mixflag), *allflag)
+}
+
+func cmdHelp(args []string) {
+	PrintMainHelp()
+}
+
+func cmdVersion(args []string) {
+	// Do nothing else since version is always printed in main.
```

Comment:
Instead of having a blank function, why don't make this actually print something - or just remove it completely since the version info is printed? I'm not against having it print the version string in both places, but I'd like to avoid empty/no-op functions.